mm2src/coins/z_coin/storage/blockdb/blockdb_sql_storage.rs

```diff
@@ -73,11 +73,13 @@ impl BlockDbImpl {
         async_blocking(move || {
             let conn = ctx
                 .sqlite_connection
-                .clone_or(Arc::new(Mutex::new(Connection::open_in_memory().unwrap())));
+                .get()
+                .cloned()
+                .unwrap_or_else(|| Arc::new(Mutex::new(Connection::open_in_memory().unwrap())));
             let conn_clone = conn.clone();
-            let conn_clone = conn_clone.lock().unwrap();
-            run_optimization_pragmas(&conn_clone).map_err(|err| ZcoinStorageError::DbError(err.to_string()))?;
-            conn_clone
+            let conn_lock = conn_clone.lock().unwrap();
+            run_optimization_pragmas(&conn_lock).map_err(|err| ZcoinStorageError::DbError(err.to_string()))?;
+            conn_lock
```

Comment:
we shouldn't need `conn_clone`. let's lock `conn` right away.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2267/commits/72018af9955a02a0bc9734ef49c02a85cab6f166https://github.com/KomodoPlatform/komodo-defi-framework/pull/2267/commits/72018af9955a02a0bc9734ef49c02a85cab6f166